pkg/dataobj/internal/dataset/value_encoding_bitmap.go

```diff
@@ -409,7 +409,8 @@ func (b *bitpackBuffer) Flush(w streamio.Writer) error {
 	//
 	// To encode the width in 6 bits, we encode width-1. That reserves the bottom
 	// 7 bits for metadata, and the remaining 57 bits for the number of sets.
-	const maxSets = 1<<57 - 1
```

Comment:
Could this be explicitly typed as `const maxSets uint64` so we wouldn't have to decrease the value on 32 bit ARM?  

(I'm fine with the other change to remove math.MinInt64)

Reply:
You compare the value later with a variable s.sets of type `int`. We could change that field to `uint64` if you want

Reply:
Haha sorry, I was reviewing this from my phone 🤦 At my desk now; I think your change is fine, we'll see if anything breaks. 

Can you update the doc comment on the bitpackBuffer struct? Right now it still says the max value for bit_packed_sets is 2^57-1.

Reply:
I changed the field `sets` to type `uint64`, which seems to be the more appropriate option.